craft_parts/state_manager/states.py

```diff
@@ -80,6 +99,17 @@ def remove(part: Part, step: Step) -> None:
         state_file.unlink()
 
 
-def state_file_path(part: Part, step: Step) -> Path:
-    """Return the path to the state file for the give part and step."""
+def step_state_path(part: Part, step: Step) -> Path:
```

Comment:
jw if we should prefix these methods with get_ or similar, but your call :D

Reply:
that would be nice even if the operation is cheap

Reply:
Renamed `step_state_path()` and `overlay_migration_state_path()` to `get_step_state_path()` and `get_overlay_migration_state_path()`.